stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/MessagesScreen.kt

```diff
@@ -224,169 +220,290 @@ public fun MessagesScreen(
+ * can use to interact with messages.
+ *
+ * @param listViewModel The [MessageListViewModel] used to read state from.
+ * @param composerViewModel The [MessageComposerViewModel] used to read state from.
+ */
+@Composable
+private fun BoxScope.MessageMenus(
+    listViewModel: MessageListViewModel,
+    composerViewModel: MessageComposerViewModel,
+) {
+    val selectedMessageState = listViewModel.currentMessagesState.selectedMessageState
 
-        if (newMessageOptions.isNotEmpty()) {
-            messageOptions = newMessageOptions
-        }
+    val selectedMessage = selectedMessageState?.message ?: Message()
 
-        AnimatedVisibility(
-            visible = selectedMessageState is SelectedMessageOptionsState && selectedMessage.id.isNotEmpty(),
-            enter = fadeIn(),
-            exit = fadeOut(animationSpec = tween(durationMillis = AnimationConstants.DefaultDurationMillis / 2))
-        ) {
-            SelectedMessageMenu(
-                modifier = Modifier
-                    .align(Alignment.BottomCenter)
-                    .animateEnterExit(
-                        enter = slideInVertically(
-                            initialOffsetY = { height -> height },
-                            animationSpec = tween()
-                        ),
-                        exit = slideOutVertically(
-                            targetOffsetY = { height -> height },
-                            animationSpec = tween(durationMillis = AnimationConstants.DefaultDurationMillis / 2)
-                        )
+    MessagesScreenMenus(
+        listViewModel = listViewModel,
+        composerViewModel = composerViewModel,
+        selectedMessageState = selectedMessageState,
+        selectedMessage = selectedMessage
+    )
+
+    MessagesScreenReactionsPicker(
+        listViewModel = listViewModel,
+        composerViewModel = composerViewModel,
+        selectedMessageState = selectedMessageState,
+        selectedMessage = selectedMessage,
+    )
+}
+
+/**
+ * Contains selected message and reactions menus
+ * wrapped inside an animated composable.
+ *
+ * @param listViewModel The [MessageListViewModel] used to read state and
+ * perform actions.
+ * @param composerViewModel The [MessageComposerViewModel] used to read state and
+ * perform actions.
+ * @param selectedMessageState The state of the currently selected message.
+ * @param selectedMessage The currently selected message.
+ */
+@Suppress("LongMethod")
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+private fun BoxScope.MessagesScreenMenus(
+    listViewModel: MessageListViewModel,
+    composerViewModel: MessageComposerViewModel,
+    selectedMessageState: SelectedMessageState?,
+    selectedMessage: Message,
+) {
+
+    val user by listViewModel.user.collectAsState()
+
+    val ownCapabilities = selectedMessageState?.ownCapabilities ?: setOf()
+
+    val isInThread = remember {
+        mutableStateOf(listViewModel.isInThread)
+    }
```

Comment:
I'm in a thread, and I can observer the "thread reply" option, which is incorrect. Could you check?

![telegram-cloud-photo-size-2-5280512207551052794-y](https://user-images.githubusercontent.com/9600921/173670850-443ebd65-b771-463f-b9c5-6b642568544e.jpg)

Reply:
According to the code snippet below the option should be omitted.

```
if (!isInThread && isMessageSynced && canReplyToMessage) {
            MessageOptionItemState(
                title = R.string.stream_compose_thread_reply,
                iconPainter = painterResource(R.drawable.stream_compose_ic_thread),
                action = ThreadReply(selectedMessage),
                titleColor = ChatTheme.colors.textHighEmphasis,
                iconColor = ChatTheme.colors.textLowEmphasis,
            )
        } else null
```